hc5/src/main/java/feign/hc5/ApacheHttp5Client.java

```diff
@@ -135,7 +135,12 @@ ClassicHttpRequest toClassicHttpRequest(Request request, Request.Options options
         entity = new ByteArrayEntity(data, null);
       } else {
         final ContentType contentType = getContentType(request);
-        entity = new StringEntity(new String(data), contentType);
+        if (request.charset() != null) {
+          String content = new String(request.body(), request.charset());
```

Comment:
can we use `data` instead?
if not, need a comment explaining why need to use `request.body()`

Reply:
Done.